.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "submodules/FileSystem"]
+	path = submodules/FileSystem
+	url = git@github.com:aspnet/FileSystem.git
```

Comment:
Submodules are never cloned for dev purpose. We won't change anything inside them and we won't push to repos in submodules. So I guess using https is fine and it is convenient by allowing you to fetch submodules without credentials.